src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,255 @@
+                        Vector<BaseFloat> *signal) {
+  if (noise->Dim()) {
+    float input_power = ComputeEarlyReverbEnergy(rir, *signal, samp_freq);
+    float noise_power = VecVec(*noise, *noise) / noise->Dim();
+    float scale_factor = sqrt(pow(10, -snr_db / 10) * input_power / noise_power);
+    noise->Scale(scale_factor);
+    KALDI_VLOG(1) << "Noise signal is being scaled with the value " << scale_factor 
+                  << " to generate output with SNR " << snr_db << "db\n";
+  }
+
+  BlockConvolveSignals(rir, signal);
+
+  if (noise->Dim() > 0) {
+    AddVectorsOfUnequalLength(*noise, signal);
+  }
+}
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+
+    const char *usage =
+        "Corrupts the wave files supplied via input pipe with the specified\n"
+        "room-impulse response (rir_matrix) and additive noise distortions\n"
+        "(specified by corresponding files).\n";
+
+    ParseOptions po(usage);
+    std::string noise_file;
+    BaseFloat snr_db = 20;
+    bool multi_channel_output = false;
+    int input_channel = 0;
+    int rir_channel = 0;
+    int noise_channel = 0;
+    bool normalize_output = true;
+    BaseFloat volume = 0;
+
+    po.Register("multi-channel-output", &multi_channel_output,
+                "Specifies if the output should be multi-channel or not");
+    po.Register("input-wave-channel", &input_channel,
+                "Specifies the channel to be used from input as only a "
+                "single channel will be used to generate reverberated output");
+    po.Register("rir-channel", &rir_channel,
+                "Specifies the channel of the room impulse response, "
+                "it will only be used when multi-channel-output is false");
+    po.Register("noise-channel", &noise_channel,
+                "Specifies the channel of the noise file, "
+                "it will only be used when multi-channel-output is false");
+    po.Register("noise-file", &noise_file,
+                "File with additive noise");
+    po.Register("snr-db", &snr_db,
+                "Desired SNR(dB) of the output");
+    po.Register("normalize-output", &normalize_output,
+                "If true, then after reverberating and "
+                "possibly adding noise, scale so that the signal "
+                "energy is the same as the original input signal.");
+    po.Register("volume", &volume,
+                "If nonzero, a scaling factor on the signal that is applied "
+                "after reverberating and possibly adding noise. "
+                "If you set this option to a nonzero value, "
+                "it will be as if you had also specified --normalize-output=false.");
+
+    po.Read(argc, argv);
+    if (po.NumArgs() != 3) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    if (multi_channel_output) {
+      if (rir_channel != 0 || noise_channel != 0)
+        KALDI_WARN << "options for --rir-channel and --noise-channel will be ignored"
```

Comment:
options for --rir-channel and --noise-channel ~~will be~~ are ignored ~~when~~ as --multi-channel-output is true."